--- libsigrok4DSL/output/csv.c	2024-11-05 02:43:10.000000000 +0000
+++ libsigrok4DSL/output/csv.c	2025-06-18 02:41:09.286013772 +0000
@@ -298,7 +298,7 @@
                     pFlagCheck++;
                 }
 
-                g_string_append_printf(*out, tmp_buffer);
+                g_string_append(*out, tmp_buffer);
             }
 
             for (j = 0; j < ctx->num_enabled_channels; j++) {
@@ -365,7 +365,7 @@
        for (i = 0; i < (uint64_t)analog->num_samples; i++) {
             ch_cfg_dex = 0;
 
-           for (j = 0; j < ch_num; j++) {
+           for (j = 0; j < (uint64_t)ch_num; j++) {
 
                if (enalbe_channel_flags[j] == 0){
                     continue;
